src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -581,6 +581,41 @@ public async Task GoAwayFrame_NewRequest_NewConnection()
             }
         }
 
+        [ConditionalFact(nameof(SupportsAlpn))]
+        public async Task GoAwayFrame_ServerDisconnect_AbortStreams()
+        {
+            using (var server = Http2LoopbackServer.CreateServer())
+            using (HttpClient client = CreateHttpClient())
+            {
+                server.AllowMultipleConnections = true;
+
+                Task<HttpResponseMessage> sendTask = client.GetAsync(server.Address);
+                Http2LoopbackConnection connection = await server.EstablishConnectionAsync();
+                int streamId = await connection.ReadRequestHeaderAsync();
+
+                await connection.SendGoAway(streamId);
+
+                // wait until GOAWAY received
+                await connection.PingPong();
+                await connection.WaitForConnectionShutdownAsync();
```

Comment:
I don't understand this. We just sent a GOAWAY that told the client that the stream is still valid and will be processed. Shouldn't the client be waiting for that request to complete? Why are we expecting it to shutdown here?

Reply:
because socket is shutdown by the server (WaitForConnectionShutdownAsync sends shutdown) so client has no reason to wait anymore. Note: this is a scenario described in https://github.com/dotnet/corefx/issues/39014

Reply:
I'll add a comment

Reply:
Ah yes, that makes sense. I was confused by the name "WaitForConnectionShutdownAsync" -- I didn't realize it would initiate a server shutdown. Adding a comment would be helpful.